### ISSUE TITLE
pathlib.Path._from_parsed_parts should call cls.__new__(cls)

### ISSUE THREAD

**Issue by 1e658ab0-5ece-486a-a504-b31bb432e87b** [python/cpython#76846]
BPO | [32665](https://bugs.python.org/issue32665)
--- | :---
Nosy | @pitrou, @pablogsal, @qb-cea
Files | <li>[reproducer.py](https://bugs.python.org/file47409/reproducer.py "Uploaded as text/plain at 2018-01-25.12:27:24 by @qb-cea"): reproducer (tested with python3.4)</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2018-01-25.12:27:25.104>
labels = ['type-bug', 'library']
title = 'pathlib.Path._from_parsed_parts should call cls.__new__(cls)'
updated_at = <Date 2018-01-27.21:38:03.770>
user = 'https://github.com/qb-cea'
```

bugs.python.org fields:
```python
activity = <Date 2018-01-27.21:38:03.770>
actor = 'pablogsal'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2018-01-25.12:27:25.104>
creator = 'qb-cea'
dependencies = []
files = ['47409']
hgrepos = []
issue_num = 32665
keywords = []
message_count = 5.0
messages = ['310671', '310758', '310888', '310889', '310890']
nosy_count = 3.0
nosy_names = ['pitrou', 'pablogsal', 'qb-cea']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue32665'
versions = ['Python 3.6']
```

</p></details>

### Linked PRs
* gh-102789

**Comment by barneygale** [comment]
#100479 removes `_from_parsed_parts()` and adds a supported route for subclassing, so would resolve this issue I believe.

**Comment by barneygale** [comment]
Fixed in Python 3.12 / #102789 / 11c302055a2aab5373c2829d1a8610a64e72ab24
